src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -571,11 +570,13 @@ protected internal override Task OnCloseAsync(TimeSpan timeout)
         protected override void OnClosed()
         {
             base.OnClosed();
-            var httpClientToDispose = _httpClient;
-            if (httpClientToDispose != null)
+            if (_httpClientCache != null && !_httpClientCache.IsDisposed)
```

Comment:
That's not possible. `OnClosed()` is called from `CommunicationObject.Close` which under a lock saves the current state and transitions it to the Closing state. Only if the previous state was Opened will OnClosed get called. Because of the lock, only a single thread will have the previous state as Opened which means OnClosed will only ever be called once. This is the only place where _httpClientCache is set to null. The null check is there to protect against the scenario of opening the ChannelFactory and closing it before a single request is made. The lock below is there to protect against concurrent modification of _httpClientCache as another thread might be trying to add an HttpClient while Dispose is being called.